### PR TITLE
Add Jenkins SCA Reachability

### PR DESCRIPTION
**Pull request by tabacmend**

### CODE REVIEW

**Comment by Dima2021** [comment]
@tabacmend The code seems redundant in the MendSCAScan.groovy file. I recommend adding the "if" statement right before the "./mend dep" execution, as the rest of the code seems to be the same.
if (reachability) {
    ./mend dep -u -r -s "*//\${JOB_NAME}//\${repo}_\${branch}" --fail-policy --non-interactive --export-results dep-results.txt
 } else {
    ./mend dep -u -s "*//\${JOB_NAME}//\${repo}_\${branch}" --fail-policy --non-interactive --export-results dep-results.txt
}

File: CI-CD/Jenkins/Mend CLI/vars/MendSCAScan.groovy [hunk 1]
```diff
@@ -1,17 +1,32 @@
...
-            export repo=$(basename -s .git $(git config --get remote.origin.url))
-            export branch=$(git rev-parse --abbrev-ref HEAD)
-            ./mend dep -u -s "*//${JOB_NAME}//${repo}_${branch}" --fail-policy --non-interactive --export-results dep-results.txt
-            if [[ "$dep_exit" == "9" ]]; then
-                  echo "[warning]  Dependency scan policy violation"
+def call(Map args = [:]) { 
+    boolean reachability = args.get('reachability', false)
+    echo 'Run Mend dependencies scan'
+    catchError(buildResult: 'SUCCESS', stageResult: 'FAILURE') {
+        if (reachability) {
```
Comment (kyallanum-MND) [inline CI-CD/Jenkins/Mend CLI/vars/MendSCAScan.groovy:5]: This if statement should only be for the ``mend dep`` command. Everything else should only be in the script once.

**Review by kyallanum-MND — CHANGES_REQUESTED** [review]
Agree with Dima. These need to be changed before merging.

**Comment by tabacmend** [comment]
@kyallanum-MND @Dima2021 
See updated script.

**Review by Dima2021 — APPROVED** [review]

**Review by kyallanum-MND — APPROVED** [review]

**Pull request merged**